code/script.js

```diff
@@ -229,97 +234,122 @@ const setSecret = () => {
+      );
     } else {
-      // alert popup that says something like: "No, the person doesnt have yellow hair! Remove all people with yellow hair"
+      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value)
+      );
+      alert(
+        `No, the person does not wear or have a ${value}! Confirm to remove all people who wear or have a ${value}.`
+      );
     }
   }
 
-  // Determine what is the category
-  // filter by category to keep or remove based on the keep variable.
-  /* 
-    for hair and eyes :
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
-
-    for accessories and other
-      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
-      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
-
-  // Invoke a function to redraw the board with the remaining people.
+  generateBoard();
 }
 
 // when clicking guess, the player first have to confirm that they want to make a guess.
 const guess = (personToConfirm) => {
-  // store the interaction from the player in a variable.
-  // remember the confirm() ?
-  // If the player wants to guess, invoke the checkMyGuess function.
+  const playerConfirmation = confirm(`Are you sure about your guess on ${personToConfirm}?`
+  );
+  if (playerConfirmation) {
+    checkMyGuess(personToConfirm);
+  } else {
+    alert(`Ok, keep guessing!`);
+  }
 }
 
 // If you confirm, this function is invoked
 const checkMyGuess = (personToCheck) => {
-  // 1. Check if the personToCheck is the same as the secret person's name
-  // 2. Set a Message to show in the win or lose section accordingly
-  // 3. Show the win or lose section
-  // 4. Hide the game board
+  if (personToCheck === secret.name) {
+    let showWinOrLose = () => {
+      document.getElementById('winOrLose').style.display = "flex"
+      winOrLoseText.innerHTML = `✨ Nailed it!!! ✨`
+    }
+    board.innerHTML = ""
+    showWinOrLose ();
+  } else {
+    let showWinOrLose = () => {
+      document.getElementById('winOrLose').style.display = "flex"
+      winOrLoseText.innerHTML = `Better luck next time, ${secret.name} was the one to look for! 👀`
+   }
+    board.innerHTML = ""
+    showWinOrLose ();
+  }
 }
 
 // Invokes the start function when website is loaded
-start()
+start();
 
 // All the event listeners
-restartButton.addEventListener('click', start)
+
+questions.addEventListener('change', selectQuestion); //when player chooses hair/eyes/accessories/other, the function selectQuestion is called
+findOutButton.addEventListener('click', checkQuestion); //when clicking on find out, the checkQuestion function is called
+restartButton.addEventListener('click', start); //restarts the game bt calling the function start
```

Comment:
Nice job with the eventlisteners!
I also checked if the restart-button at your game works and it does! Both after some guesses or also if you guess on the wrong person. GREAT JOB!

I saw a minor thing that I think has to do with the CSS that you could fix if there's time (I'll try to post the printscreen here but if it doesn't work I'll send it on Slack instead :) ) 
![image](https://user-images.githubusercontent.com/121539816/220782784-87d78217-9cdc-4c53-80b5-240a765f89bb.png)

After finishing the game - the game section is still there on the left side - I think you can chose the win-or-lose-section (or what you call it in your code) to cover the whole screen so that the game section is not visible. Hope that the pic I put her works so you see what I mean :)